.github/pull_request_template.md

```diff
@@ -0,0 +1,50 @@
+# Description
+Mandatory, single line description of the pull request, written as though it was an order.
+
+Then, following a single blank line, a summary of the change and which issue is fixed, including 
+- [ ] what is the problem being solved
+- [ ] why this is the best approach
+- [ ] what are the shortcomings of this approach, if any
+It may include
+- [ ] background information, such as "Fixes #"
+- [ ] benchmark results
+- [ ] links to design documents
+- [ ] dependencies required for this change
+
+The summary may omit some of these details if this PR fixes a single ticket that includes these details. It is the reviewer's discretion. 
+
+## [Optional] Changes
+Multiple changes are not recommended, so this section should normally be omitted. Unforunately, they are sometimes unavoidable. If there are multiple logical changes, list them separately.
+
+1. `'foo'` is replaced by `'bar'`
+2. `'fizzbuzz'` is optimized for gas
+
+## Type of change
+
+Feel free to delete options that are not relevant.
+
+- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
+- [ ] Bug fix (non-breaking change which fixes an issue)
+- [ ] New feature (non-breaking change which adds functionality)
+- [ ] Chore (tech debt, improved developer experience, etc)
+- [ ] Performance improvement (either gas efficiency or computational efficiency)
+- [ ] Requires a documentation update
+
+# [Optional] How Has This Been Tested?
+
+Did you need to run manual tests to verify this change? If so, please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
+
+# Checklist:
+
+## Code quality
+- [ ] I have written clear commit messages
+- [ ] I have performed a self-review of my own code
+- [ ] I have scoped this change as narrowly as possible
+- [ ] I have separated functional and non-functional changes
+- [ ] I have commented my code strictly where necessary
```

Comment:
Curious about this one. 

1) I guess you mean "only where necessary", rather than "strictly, wherever necessary"?
2) if yes to 1, is this the right policy? I think there is an argument that good commenting should be encouraged.

Reply:
I didn't know how to phrase it to encourage exactly what you suggest.

Reply:
Personally, I'd prefer something like `Wherever necessary I have commented my code`.

I'd rather lean towards too many comments then too few

Reply:
Is this ok?

```suggestion
- [ ] I have commented my code wherever necessary
```